[HPB-190] [DevOps] Stage 環境建立

JYu1999 opened the pull request:
https://www.notion.so/jyu1999/DevOps-Stage-294cba948f1180b1a2a5c9febb345599?source=copy_link

此 PR 需連同以下 PR 一起 Review:
- https://github.com/Goodidea-backend-camp/hpb-blog-project/pull/7
- https://github.com/Goodidea-backend-camp/hpb-blog-backend/pull/9

JYu1999 committed:
feat(nuxt.config): enhance configuration for environment-specific allowed hosts and server settings
- Introduced `getAllowedHosts` function to dynamically set allowed hosts based on the environment (dev, stage, prod).
- Added `environment` to runtime configuration and updated Vite server settings to include dynamic host and port configuration.

JYu1999 committed:
fix(index.vue): update API endpoint for health check
- Changed the fetch URL from '/api/' to '/api/healthz' to target the health check endpoint.

JYu1999 committed:
chore(Dockerfile.dev): remove development Dockerfile

JYu1999 committed:
feat(nuxt.config): configure development server settings
- Added `devServer` configuration to set host to '0.0.0.0' and port to 3000 for local development.
- Removed redundant server settings from Vite configuration to streamline setup.

richardlinio commented on nuxt.config.ts:
目前我看 main 分支，這一段在上次 HPB-162 合進來後如下

```
routeRules: {
    // TEMPORARILY DISABLED: Using Nitro dev server mock APIs instead
    // Uncomment when backend server is ready
    // '/api/**': {
    //   proxy: `http://backend:${backendPort}/**`
    // },
    '/admin/**': {
      ssr: false
    }
  },
```

這邊 uncomment 後的設定是對的，但 GitHub 沒出現紅紅綠綠的增刪記號，是否結語本地創建 HPB-190 的時候還沒有 git pull main，或是尚未 rebase 呢？

這裡不會有什麼危害，頂多是解衝突

JYu1999 replied:
https://github.com/Goodidea-backend-camp/hpb-blog-frontend/pull/10/commits/9c8855335746cb8db8d9378586eb694e9a0a3a90

忘記了，感謝提醒

richardlinio replied:
感謝～

既然現在要建立環境，代表前後端要接通，那這裡我們會需要 uncomment，讓 Nuxt 處理這個 proxy。
```
'/api/**': {
     proxy: `http://backend:${backendPort}/**`
   },
```

這樣就不會運作假資料，但基於我們改成先開發後端，可以預期前端需要假資料的機率將降低，我認為就 uncomment 吧

JYu1999 committed:
Merge branch 'main' into feature/HPB-190

richardlinio commented on nuxt.config.ts:
既然現在要建立環境，代表前後端要接通，那這裡我們會需要 uncomment，讓 Nuxt 處理這個 proxy。
```
'/api/**': {
     proxy: `http://backend:${backendPort}/**`
   },
```
這樣就不會運作假資料，但基於我們改成先開發後端，可以預期前端需要假資料的機率將降低，我認為就 uncomment 吧

richardlinio replied:
這邊如果沒有 uncomment，我們首頁的 /healthz 一定會失敗

JYu1999 replied:
https://github.com/Goodidea-backend-camp/hpb-blog-frontend/pull/10/commits/8598e17e7d907f0566660757c75238d0efaadebc

JYu1999 committed:
fix(nuxt.config): re-enable backend proxy for API routes
- Uncommented the proxy configuration for '/api/**' routes to connect to the backend server, previously disabled for development purposes.

JYu1999 committed:
feat(nuxt.config): enhance allowed hosts configuration based on environment variables
- Updated `getAllowedHosts` function to read allowed hosts from environment variables, providing defaults based on the current environment if not set. This improves flexibility for different deployment scenarios.
